http-service/src/main/java/net/runelite/http/service/feed/FeedController.java

```
@@ -0,0 +1,114 @@
+
+import com.google.common.base.Suppliers;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.concurrent.TimeUnit;
+import java.util.function.Supplier;
+import net.runelite.http.api.feed.FeedResult;
+import net.runelite.http.api.feed.FeedItem;
+import net.runelite.http.service.feed.blog.BlogService;
+import net.runelite.http.service.feed.osrsnews.OSRSNewsService;
+import net.runelite.http.service.feed.twitter.TwitterService;
+import net.runelite.http.service.util.exception.InternalServerErrorException;
+import net.runelite.http.service.util.exception.RemoteServerErrorException;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+
+@RestController
+@RequestMapping("/feed")
+public class FeedController
+{
+	private static final Logger logger = LoggerFactory.getLogger(FeedController.class);
+
+	private BlogService blogService;
+	private TwitterService twitterService;
+	private OSRSNewsService osrsNewsService;
+
+	private final Supplier<FeedResult> feed = Suppliers.memoizeWithExpiration(() ->
+	{
+		List<FeedItem> items = new ArrayList<>();
+
+		try
+		{
+			/*
+				Don't break if one of the services are unavailable
+			 */
+
+			try
+			{
+				items.addAll(blogService.getBlogPosts());
+			}
+			catch (IOException | RemoteServerErrorException e)
+			{
+				logger.warn(null, e);
+			}
+
+			try
+			{
+				items.addAll(twitterService.getTweets());
+			}
+			catch (IOException | RemoteServerErrorException e)
+			{
+				logger.warn(null, e);
+			}
+
+			try
+			{
+				items.addAll(osrsNewsService.getNews());
+			}
+			catch (IOException | RemoteServerErrorException e)
+			{
+				logger.warn(null, e);
+			}
+		}
+		catch (InternalServerErrorException e)
+		{
+			throw new RuntimeException();
```

Don't swallow exceptions